core/src/main/java/org/infinispan/container/entries/ClusteredRepeatableReadEntry.java

```diff
@@ -82,7 +79,7 @@ private EntryVersion getCurrentEntryVersion(DataContainer container, Persistence
       EntryVersion prevVersion;// on origin, the version seen is acquired without the lock, so we have to retrieve it again
       // TODO: persistence should be more orthogonal to any entry type - this should be handled in interceptor
       InternalCacheEntry ice = PersistenceUtil.loadAndStoreInDataContainer(container, persistenceManager, getKey(),
-            ctx, timeService, ignored);
+            ctx, timeService, null);
```

Comment:
I haven't benchmarked it, but volatile write to single location would limit scalability. Volatile write requires to acquire exclusive ownership of the cache line, and I smell contention here. Theoretically, the check could be slower in an uncontended case (though I doubt so), but once we start hitting the reference with many threads...